client/src/game/mod.rs

```diff
@@ -30,6 +30,10 @@ pub fn start_game(
     let mut game = ClashGame;
     let mut player_id = 0;
     let mut lobby = None;
+    // Not sure if this is the best approach, the idea is that it would be faster
+    // to store a local state of what we as a client have collected rather
+    // than searching to see if we've collected it.
+    let mut local_spat_state = HashSet::<Spatula>::with_capacity(Spatula::COUNT);
```

Comment:
Haven't reviewed too closely yet. Will this cause issues if a player collects a spatula but the server doesn't respect the collection (such as if it's only available one more time and two players collect it very close to each other)? One of the benefits of having all our state on the NetworkedLobby is that we know that our state always reflects the servers true state.